Libplanet.Net/Consensus/ConsensusContext.Event.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using Libplanet.Action;
+using Libplanet.Net.Messages;
+
+namespace Libplanet.Net.Consensus
+{
+    public partial class ConsensusContext<T>
+        where T : IAction, new()
+    {
+        /// <inheritdoc cref="Context{T}.ExceptionOccurred"/>
+        internal event EventHandler<Exception>? ExceptionOccurred;
+
+        /// <inheritdoc cref="Context{T}.TimeoutOccurred"/>
+        internal event EventHandler<(Step Step, TimeSpan TimeSpan)>? TimeoutOccurred;
+
+        /// <inheritdoc cref="Context{T}.TimeoutProcessed"/>
+        internal event EventHandler<int>? TimeoutProcessed;
+
+        /// <inheritdoc cref="Context{T}.StateChanged"/>
+        internal event EventHandler<(int MessageLogSize, int Round, Step Step)>? StateChanged;
+
+        /// <inheritdoc cref="Context{T}.MessageConsumed"/>
+        internal event EventHandler<ConsensusMessage>? MessageConsumed;
+
+        /// <inheritdoc cref="Context{T}.MutationConsumed"/>
+        internal event EventHandler<System.Action>? MutationConsumed;
+
```

Comment:
This loses context, i.e. height information. I was thinking more of having height info added here.

Something along the lines of

```csharp
internal event EventHandler<(long Height, int MessageLogSize, int Round, Step Step)>? StateChanged;
```

and

```csharp
context.StateChanged += (sender, size, round, step) => StateChanged?.Invoke(sender, height, size, round, step);
```

Reply:
Amended. Seems there is no good way to concatenate the tuple, so let it handle two named tuples.
https://github.com/planetarium/libplanet/pull/2383/commits/22a6ea572d7217f318f78c5c8ff403a6e579f822#diff-d5fcd6416d32a07c2ab500ee36b81596dfd297f498c1c4d691040141b5c1ebd3